### DISCUSSION TITLE
Can't query JSONB column

### DISCUSSION THREAD

**Discussion by christian-valadez** [supabase/supabase#2102] [Questions]
I have a users table with 2 columns - user_id, and metadata. Metadata is a JSONB column that has an array of objects, and each object has an "item_id" and "access_token" value. 

```
[ { item_id: "1ab", access_token: "2bc }, { item_id: "3ab", access_token: "4bc" } ] 
```
 
 I'm trying to find the rows with a given item_id, but Supabase keeps returning an empty array. 
 
 ```
  await supabase
		.from('users')
		.select()
		.eq('metadata->>item_id', '1ab');
```
 
Any insight would be greatly appreciated!

**Comment by easy-street** [comment]
[`contains`](https://supabase.io/docs/reference/javascript/contains) supports `jsonb` values with an array shape:
```
await supabase
  .from('users')
  .select()
  .contains('metadata', '[{"item_id":"1ab"}]');
```

**Comment by inian** [comment]
I think this should work too @christian-valadez 
```
 await supabase
   	.from('users')
   	.select()
   	.eq('metadata->item_id', '1ab');
```

**Comment by vidicunt** [comment]
Neither of these solutions work for me, I am being returned an empty list
